### ISSUE TITLE
Extend the concept of showing trees to association relationships

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbemtools#19]
We have a display for class hiearchies:

```
-CIM_System
 +-- CIM_ApplicationSystem
 |   +-- CIM_J2eeServer
 |   +-- CIM_J2eeApplication
 |   +-- CIM_DatabaseSystem
 +-- CIM_ComputerSystem
 |   +-- CIM_VirtualComputerSystem
 |   +-- CIM_Cluster
 |   +-- CIM_UnitaryComputerSystem
 |       +-- PG_ComputerSystem
 +-- CIM_AdminDomain
 |   +-- CIM_RoutingProtocolDomain
 |   |   +-- CIM_OSPFArea
 |   +-- CIM_ReusablePolicyContainer
 |   +-- CIM_Network
 |   +-- CIM_PolicyRepository
 |   +-- CIM_J2eeDomain
 |   +-- CIM_AutonomousSystem
 +-- CIM_StorageLibrary
```
It would be logical to extend this concept to display association relationships also.

DISCUSSION: Today the class hierarchy is shown with a special command (class hierarchy) rather than a display output option.  This was done because a: it was easier to develop isolated. b: it includes an option that enumerates do not include (it can display up the hierarchy as well as down)
 
```
class hierarchy CIM_System -s

-root
 +-- CIM_ManagedElement
     +-- CIM_ManagedSystemElement
         +-- CIM_LogicalElement
             +-- CIM_EnabledLogicalElement
                 +-- CIM_System
```

Would like help to determine what is the best way to integrate this into the tool and not sure that a special command is the easiest (especially since hierarchy is not easy to spell)

However, in any case it would be logical to have something to also display the relationships between source, references, targets rather than just displaying lists as the existing references and associators commands do.

See pr #177

Here is an example of the current tree format of   a shrub of CIM_ProtocolEndpoint.

```
instance shrub CIM_ProtocolEndpoint.?

...
CIM_SCSIProtocolEndpoint.Name="ACME+0_0_0",SystemCreationClassName="CIM_ComputerSystem",SystemName="ACME+CF2A5091300089",CreationClassName="CIM_SCSIProtocolEndpoint"
 +-- Target
     +-- CIM_SCSIInitiatorTargetLogicalUnitPath
         +-- Initiator
         |   +-- CIM_SCSIProtocolEndpoint(2 insts)
         |   |   +-- CIM_SCSIProtocolEndpoint.Name="ACME+CF2A5091300089+SP_A+12",SystemCreationClassName="CIM_ComputerSystem",SystemName="ACME+CF2A5091300089+SP_A",CreationClassName="CIM_SCSIProtocolEndpoint"
         |   |   +-- CIM_SCSIProtocolEndpoint.Name="ACME+CF2A5091300089+SP_B+12",SystemCreationClassName="CIM_ComputerSystem",SystemName="ACME+CF2A5091300089+SP_B",CreationClassName="CIM_SCSIProtocolEndpoint"
         |   +-- CIM_StorageExtent(0 insts)
         +-- LogicalUnit
             +-- CIM_SCSIProtocolEndpoint(0 insts)
             +-- CIM_StorageExtent(1 insts)
                 +-- CIM_StorageExtent.SystemCreationClassName="CIM_ComputerSystem",SystemName="ACME+CF2A5091300089",CreationClassName="CIM_StorageExtent",DeviceID="ACME+CF2A5091300089+2;3;;0_0_0;;;0;0;;"
```

And a an example showing both the tree format and table format for the mock association used in
our tests.

```
Input integer between 0 and 7 or Ctrl-C to exit selection: : 0
TST_Person.name="Mike"
 +-- parent
 |   +-- TST_Lineage
 |       +-- child
 |           +-- TST_Person(2 insts)
 |               +-- TST_Person.name="Sofi"
 |               +-- TST_Person.name="Gabi"
 +-- member
     +-- TST_MemberOfFamilyCollection
         +-- family
             +-- TST_FamilyCollection(1 insts)
                 +-- TST_FamilyCollection.name="Family2"
pywbemcli> -o table instance shrub TST_Person.?
Pick Instance name to process
0: root/cimv2:TST_Person.name="Mike"
1: root/cimv2:TST_Person.name="Saara"
2: root/cimv2:TST_Person.name="Sofi"
3: root/cimv2:TST_Person.name="Gabi"
4: root/cimv2:TST_PersonSub.name="Mikesub"
5: root/cimv2:TST_PersonSub.name="Saarasub"
6: root/cimv2:TST_PersonSub.name="Sofisub"
7: root/cimv2:TST_PersonSub.name="Gabisub"
Input integer between 0 and 7 or Ctrl-C to exit selection: : 0
Shrub of root/cimv2:TST_Person.name="Mike"
+--------+------------------------------+--------------+----------------------+-------------------------------------+
| Role   | Reference Classes            | ResultRole   | Associated Classes   | Assoc Inst paths                    |
|--------+------------------------------+--------------+----------------------+-------------------------------------|
| parent | TST_Lineage                  | child        | TST_Person           | TST_Person.name="Sofi"              |
|        |                              |              |                      | TST_Person.name="Gabi"              |
| member | TST_MemberOfFamilyCollection | family       | TST_FamilyCollection | TST_FamilyCollection.name="Family2" |
+--------+------------------------------+--------------+----------------------+-------------------------------------+
pywbemcli>
```

**Comment by KSchopmeyer** [comment]
**DISCUSSION** Alternatives for display of tree.

```
Proposed formats for tree display
============================================================ 0
Current display - role, AsocClass, ResultRole, ResultClass, ResultInstances
each has tree level

Pick Instance name to process
0: root/cimv2:TST_EP.InstanceID=1
1: root/cimv2:TST_EP.InstanceID=2
Input integer between 0 and 1 or Ctrl-C to exit selection: : 0
TST_EP.InstanceID=1
 +-- Initiator(Role)
     +-- TST_A3(AssocClass)
         +-- Target(ResultRole)
         |   +-- TST_EP(ResultClass)(1 insts)
         |       +-- TST_EP.InstanceID=2
         +-- LogicalUnit(ResultRole)
             +-- TST_LD(ResultClass)(2 insts)
             |   +-- TST_LD.InstanceID=3
             |   +-- TST_LDSub.InstanceID=4
             +-- TST_LDSub(ResultClass)(1 insts)
                 +-- TST_LDSub.InstanceID=4


=============================================================== 2
Combine Role and AssocClass

TST_EP.InstanceID=1
 +-- Initiator(Role):TST_A3(AssocClass)
     +-- Target(ResultRole)
     |   +-- TST_EP(ResultClass)(1 insts)
     |       +-- TST_EP.InstanceID=2
     +-- LogicalUnit(ResultRole)
         +-- TST_LD(ResultClass)(2 insts)
         |   +-- TST_LD.InstanceID=3
         |   +-- TST_LDSub.InstanceID=4
         +-- TST_LDSub(ResultClass)(1 insts)
             +-- TST_LDSub.InstanceID=4

=================================================================== 3
Don't show indent for ResultClasses; Just show instances under ResultRole

TST_EP.InstanceID=1
 +-- Initiator(Role)
     +-- TST_A3(AssocClass)
         +-- Target(ResultRole)
         |   +-- TST_EP.InstanceID=2
         +-- LogicalUnit(ResultRole)
             +-- TST_LD.InstanceID=3
             +-- TST_LDSub.InstanceID=4
             +-- TST_LDSub.InstanceID=4

==========================================================  4
Combmines 2 and 3 above. This is most compact
TST_EP.InstanceID=1
 +-- Initiator(Role):TST_A3(AssocClass)
     +-- Target(ResultRole)
         +-- TST_EP.InstanceID=2
     +-- LogicalUnit(ResultRole)
         +-- TST_LD.InstanceID=3
         +-- TST_LDSub.InstanceID=4
         +-- TST_LDSub.InstanceID=4

=============================================================
I

**Comment by andy-maier** [comment]
Karl and I agreed on a way to represent that, which is implemented in PR #177 but not documented above. Need to review the output of that implementation.

**Comment by andy-maier** [comment]
PR #177 addresses this in master.
